napari/_vispy/layers/base.py

```diff
@@ -117,8 +119,26 @@ def _on_visible_change(self):
     def _on_opacity_change(self):
         self.node.opacity = self.layer.opacity
 
-    def _on_blending_change(self):
-        blending_kwargs = BLENDING_MODES[self.layer.blending]
+    def _on_blending_change(self, event=None):
+        blending = self.layer.blending
+        blending_kwargs = BLENDING_MODES[blending].copy()
+
+        if self.first_visible:
+            # if the first layer, then we should blend differently
+            # (ignore the canvas color but allow alpha blending)
+            src_color_blending = (
+                'src_alpha'
+                if blending not in ('minimum', 'additive')
+                else 'one'
+            )
+            blending_kwargs = dict(
+                depth_test=True,
+                cull_face=False,
+                blend=True,
+                blend_func=(src_color_blending, 'zero', 'one', 'one'),
```

Comment:
```suggestion
                blend_func=(src_color_blending, 'one_minus_src_alpha', 'one', 'one'),
```
I think this is needed to prevent black box when blending a src with 0 alpha.
This way if src has 0 alpha, then the src color is zeroed out and canvas color is used with full alpha.
This makes sense I think.

Reply:
Nice catch; I'm thinking that maybe this needs to actually be different for `minimum` and `additive` like for `src_color_blending`? Otherwise I *think* if you have an `additive` stack with non-black canvas, you'd get leakage if you change opacity of the bottommost layer?

```py
            dst_color_blending = (
                'one_minus_src_alpha'
                if blending not in ('minimum', 'additive')
                else 'zero'
            )
```

Reply:
Can you test the last commit?

Reply:
To me the new behavior is still not right.
Try an image like File > Samples > Cell
Try to adjust opacity: no matter theme, it does nada (theme still plays a role
Use Light theme and make a Labels layer: black square.
Try Cells3D and put labels at the bottom—corner case, sure. Use Light theme and make both of the Cells layers `minimum` with one of the new inverted caps. On live this works fine, with this PR it's a black box.
I don't think it's possible to have a concept of `translucent` with alpha control for the bottom-most visible—or single—layer without taking into account the canvas.

In playing around with it, I get the most intuitive behavior when I comment out 
`        color.alpha = 0`
And:
```
if self.first_visible:
            # if the first layer, then we should blend differently
            # (ignore the canvas color but allow alpha blending)
            # if blending in ('minimum', 'additive'):
            #     src_color_blending = 'src_alpha'
            #     dst_color_blending = 'one_minus_src_alpha'
            # else:
            #     src_color_blending = 'one'
            #     dst_color_blending = 'zero'
            blending_kwargs = dict(
                depth_test=True,
                cull_face=False,
                blend=True,
                blend_func=(
                    'src_alpha',
                    'one_minus_src_alpha',
                    'one',
                    'one',
                ),
                blend_equation='func_add',
            )
```
This basically treats bottom-most-visible as a `translucent` layer. if you make it opaque (alpha = 1), which is the default, it eliminates blending with canvas. I think there is an expectation that if you make it less opaque, you will see an effect. So I think you have to blend with canvas, but at that point you control the effect
Just the `minimum`+Dark theme+bottom-most-visible is shapes/points/labels makes a black box, but I think that's really (corner case)^2 because those layers don't have their own cmap/background.

2nd caveat is if the bottom most layer is something like`additive` (or with my suggestion `translucent`) and 0.7 opacity and then switch to `minimum`. Obviously, we override `minimum`, so then the opacity *can be used*, but cannot be changed, because `minimum` setting overrides the slider. Again a real corner case.

Reply:
Damn, so sorry, I actually inverted that if statement at some point during committing :/ It should be:

```py
            if blending in ('minimum', 'additive'):
                src_color_blending = 'one'
                dst_color_blending = 'zero'
            else:
                src_color_blending = 'src_alpha'
                dst_color_blending = 'one_minus_src_alpha'
```

This is so that no blending with the canvas happens with additive and minimum (so you have full black or full white as the "background" color.

Reply:
Gah, I should have caught that—sorry!
Works **super** nice now and makes **a lot** more sense!

Reply:
(Just for the record, the shapes/points/labels at the bottom of a stack with `minimum` & Dark theme does result in a black box, but pretty sure as mentioned about that requires letting those layers set their own background)

Reply:
Awesome! Yup, and I think we can leave that out of this PR for now, but let's keep it in mind if this issue comes up :)